### PR TITLE
[4.0] Smart Search: Adding debug information

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
When running the indexer of Smart Search in the backend, the output is very unsatisfying for debugging. When debugging is enabled, this PR adds a list of all contexts that are being indexed with the total number of items to index and the currently indexed items.

### Testing Instructions
1. Apply PR
2. Run ```npm i```
3. Enable debugging in global configuration
4. Run the indexer of Smart Search in the backend and see the

### CODE REVIEW

**Comment by laoneo** [comment]
Can you make the dog here happy?

**Comment by Hackwar** [comment]
Unfortunately no. I don't know the best solution for these issues and if you could help me with this, I'd be very gratefull.

**Comment by wilsonge** [comment]
Woof is happy.

https://eslint.org/docs/rules/prefer-template (formatting strings
https://eslint.org/docs/rules/guard-for-in (you were missing the `hasOwnProperty` on one loop and on the other on a style eslint didn't like)

Finally we're using the airbnb style guide which chooses to disallow for in loops because of the whole prototype thing https://github.com/airbnb/javascript#iterators--nope 

To fix that i used `Object.entries` which isn't the most efficient ever (Object.keys is probably better looking for 10 seconds at https://hackernoon.com/5-techniques-to-iterate-over-javascript-object-entries-and-their-performance-6602dcb708a8 ). But I wasn't sure exactly what the data structure was as was just doing this from github editor (if we had key => value or just a list)

**Comment by TobsBobs** [comment]
I have tested this item :red_circle: unsuccessfully on 9d63258e4c1f3e2123a0cd5a4be26f07ecec1638<br /><br />

I apply this patch and run npm install. Without this patch the indexer runs correctly. With patch applied the indexer don´t run.

With patch:


Without patch:



<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22593">issues.joomla.org/tracker/joomla-cms/22593</a>.</sub>

**Comment by Hackwar** [comment]
I've gone through this again and hopefully fixed it. Can you check again?

**Comment by TobsBobs** [comment]
I have tested this item :white_check_mark: successfully on b042459093e83ddcd725a9f96be8ebc117c08e47<hr />

unbenannt

<sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22593">issues.joomla.org/tracker/joomla-cms/22593</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b042459093e83ddcd725a9f96be8ebc117c08e47<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22593">issues.joomla.org/tracker/joomla-cms/22593</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22593">issues.joomla.org/tracker/joomla-cms/22593</a>.</sub>

File: build/media_src/com_finder/js/indexer.es6.js [hunk 1]
```diff
@@ -76,9 +76,22 @@
         if (json.start) {
           // eslint-disable-next-line prefer-destructuring
           totalItems = json.totalItems;
+          if (document.getElementById('finder-debug-data')) {
+            const debuglist = document.getElementById('finder-debug-data');
+            Object.entries(json.pluginState).forEach((context) => {
+              let item = `<dt class="col-sm-3">${context[0]}</dt>`;
```
Comment (laoneo) [inline build/media_src/com_finder/js/indexer.es6.js:82]: Bootstrap classes shouldn't be used in Javascript. Can this be done in a different way?

Comment (Hackwar) [reply build/media_src/com_finder/js/indexer.es6.js:82]: In theory I agree with you, in practice this is only used in the backend and in this specific area. I would call this okay.

**Pull request merged**